resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala

```diff
@@ -59,11 +59,16 @@ private[spark] class BasicExecutorFeatureStep(
   private val isDefaultProfile = resourceProfile.id == ResourceProfile.DEFAULT_RESOURCE_PROFILE_ID
   private val isPythonApp = kubernetesConf.get(APP_RESOURCE_TYPE) == Some(APP_RESOURCE_TYPE_PYTHON)
   private val disableConfigMap = kubernetesConf.get(KUBERNETES_EXECUTOR_DISABLE_CONFIGMAP)
+  private val memoryOverheadFactor = if (kubernetesConf.contains(EXECUTOR_MEMORY_OVERHEAD_FACTOR)) {
+    kubernetesConf.get(EXECUTOR_MEMORY_OVERHEAD_FACTOR)
+  } else {
+    kubernetesConf.get(MEMORY_OVERHEAD_FACTOR)
+  }
```

Comment:
https://github.com/apache/spark/blob/cd86df881c9570d32f6522ee163b884fda00a530/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicDriverFeatureStep.scala#L62-L63

I found it, it is propagated to executors from driver conf

Reply:
yeah thanks this is what I was also just looking at but I'm not sure how it was propagated to the executors. I was looking at through the KubernetesDriverconf somehow or possible through the pod system properties:
 MEMORY_OVERHEAD_FACTOR.key -> overheadFactor.toString.

If you find it let me know, still investigating.

Reply:
ok, I think I see how this is happening:

```
 val resolvedDriverSpec = builder.buildFromFeatures(conf, kubernetesClient)
    val configMapName = KubernetesClientUtils.configMapNameDriver
    val confFilesMap = KubernetesClientUtils.buildSparkConfDirFilesMap(configMapName,
      conf.sparkConf, resolvedDriverSpec.systemProperties)
```

We build the driver spec, which includes the added system properties:

`spec.systemProperties ++ addedSystemProperties`

Added system properties in driver feature steps add the memory overhead setting there:
```

val additionalProps = mutable.Map(
      KUBERNETES_DRIVER_POD_NAME.key -> driverPodName,
      "spark.app.id" -> conf.appId,
      KUBERNETES_DRIVER_SUBMIT_CHECK.key -> "true",
      MEMORY_OVERHEAD_FACTOR.key -> overheadFactor.toString)
```

Reply:
so is it that the system property added now is `spark.driver.memoryOverheadFactor` so the executor ignores it?

Reply:
Should/can the Executor feature step add it's own memory overhead or do we need to also calculate the executor overhead in the driver feature step and added it as a system prop?

Reply:
@Kimahriman do you have time to put up PR to fix?

Reply:
Still a little confused though, where/how is `EXECUTOR_MEMORY_OVERHEAD_FACTOR` now used for k8s or is it just ignored in favor of whatever is passed from the driver?

Reply:
Hi, @tgravescs and @mridulm . Yes, we need to fix this, but before that I'm wondering if we really need this at Apache Spark 3.3. As we noticed here, it turns out this is not a straight forward contribution. If you don't mind, I'd like to recommend to keep fixing and stabilizing in `master` branch while reverting this from `branch-3.3` at least.

Reply:
in my mind this is a small bug and it is easily fixed. I disagree with not a straight forward contribution.  If you want to revert in branch 3.3 go ahead and we can just remerge when fixed

@Kimahriman logic in ExecutorFeatureSTeps stays the same.  EXECUTOR_MEMORY_OVERHEAD_FACTOR is used if its specified, otherwise it falls back to whatever MEMORY_OVERHEAD_FACTOR is and in this case the only difference is DriverFeatureStep takes care of setting it to the default when its not explicitly specified by user.

Reply:
I should also state that if people don't want in 3.3, I'm personally fine with it just need input from anyone interested in the feature.

Reply:
@Kimahriman sorry, I guess there is one correction to what I say, we only want MEMORY_OVERHEAD_FACTOR to reflect what it would be if it wasn't set via DRIVER_MEMORY_OVERHEAD_FACTOR

Reply:
I realized that might get ugly, I'll put up a pr shortly

Reply:
> I should also state that if people don't want in 3.3, I'm personally fine with it just need input from anyone interested in the feature.

Thank you. Then, it's simpler because this PR was backported manually after feature freeze. :)
We are currently discussing on the whitelisting about late arrivals like this PR, aren't we, @tgravescs ?
We can discuss there together to get those input you need.

Reply:
Okay I think I get it, those "system properties" end up as default spark configs on the executor. Clear as mud

Reply:
https://github.com/apache/spark/pull/35900 revert pr

Reply:
Thank you for your decision, @tgravescs .